src/isolate.rs

```diff
@@ -37,23 +39,22 @@ pub type Op = Future<Item = Buf, Error = DenoError> + Send;
 
 // Returns (is_sync, op)
 pub type Dispatch =
-  fn(state: Arc<IsolateState>, buf: &[u8], data_buf: &'static mut [u8])
+  fn(isolate: &mut Isolate, buf: &[u8], data_buf: &'static mut [u8])
     -> (bool, Box<Op>);
 
 pub struct Isolate {
   ptr: *const libdeno::isolate,
   dispatch: Dispatch,
   rx: mpsc::Receiver<(i32, Buf)>,
   ntasks: i32,
+  pub timeout_due: Option<Instant>,
   pub state: Arc<IsolateState>,
 }
 
 // Isolate cannot be passed between threads but IsolateState can. So any state that
 // needs to be accessed outside the main V8 thread should be inside IsolateState.
 pub struct IsolateState {
-  pub timeout: AtomicIsize,
```

Comment:
I see. Im guessing your change above to how msg_from_js dispatches is because you want to optimize away this atomic variable

Reply:
Call it "just optimize" if you will, but I think there's also a communication in what data we use.
Just like `ntasks`, `timeout` cannot be safely read or written from another thread. 
So it needs to be in Isolate and not IsolateState.

Reply:
I agree.